### PR TITLE
[JIRA GEOS-8899] [Community Security Plugins] Include Keycloak security plugins artifacts to community dependencies and release

### PR DESCRIPTION
**Pull request by afabiani**
closes https://osgeo-org.atlassian.net/browse/GEOS-8899

### CODE REVIEW

File: src/community/release/ext-geofence-server.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
         <include>*geofence-services*</include>
         <include>*geofence-persistence*</include>
         <include>hatbox*</include>
-        <include>*hibernate*</include>
+        <include>hibernate*</include>
```
Comment (aaime) [inline src/community/release/ext-geofence-server.xml]: Is this unrelated? Or keycloack brings in hibernate jars that should not be in GeoFence?

File: src/community/security/keycloak/src/main/java/org/geoserver/security/keycloak/AuthResults.java [hunk 2]
```diff
@@ -49,16 +49,24 @@ public AuthResults(AuthChallenge challenge) {
      * @param authentication valid credentials
      */
     public AuthResults(Authentication authentication) {
-        assert authentication.getDetails() instanceof SimpleKeycloakAccount;
-        final SimpleKeycloakAccount details = (SimpleKeycloakAccount) authentication.getDetails();
+        Object username = null;
+        Object details = null;
+        if (authentication.getDetails() instanceof SimpleKeycloakAccount) {
```
Comment (aaime) [inline src/community/security/keycloak/src/main/java/org/geoserver/security/keycloak/AuthResults.java]: Is there a case where we are using a keycloak plugin and not getting a SimpleKeycloakAccount?

File: src/community/release/ext-geofence-server.xml (hunk 1 above)
Comment (afabiani) [reply src/community/release/ext-geofence-server.xml]: Exactly. Written like this the geofence will import <evrything>-hibernate which might cause issues to the keycloak plugin if not correctly configured. Typical example is the gs-monitoring-hibernate.jar which is wrongly included into the geofence-server package and therefore causing issues to security plugins too.

File: src/community/security/keycloak/src/main/java/org/geoserver/security/keycloak/AuthResults.java (hunk 2 above)
Comment (afabiani) [reply src/community/security/keycloak/src/main/java/org/geoserver/security/keycloak/AuthResults.java]: Yeps, when a user has logged in through the form method and the security context is already available. The plugin checks first if there's a principal already present into the security context with a matching username. In that case it brings the roles from the latter.

**Comment by afabiani** [comment]
@aaime any other question on this? Can I merge it?

**Comment by aaime** [comment]
No objections

**Pull request merged**